SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -2925,17 +2929,18 @@ void CSolver::Read_SU2_Restart_Binary(CGeometry *geometry, const CConfig *config
                    string("possible with the READ_BINARY_RESTART option."), CURRENT_FUNCTION);
   }
 
-  /*--- Store the number of fields to be read for clarity. ---*/
+  /*--- Store the number of fields and points to be read for clarity. ---*/
 
-  nFields = Restart_Vars[1];
+  const unsigned long nFields = Restart_Vars[1];
+  const unsigned long nPointFile = Restart_Vars[2];
```

Comment:
Any specific reason to take the unsigned long datatype here for nFields? Imo that is not necessary as the number of fields is typically <100. And Restart_Vars consists of int's 
That also applies to `nPointFile` which replaces `geometry->GetnPointDomain()` which actually returns an unsigned long ... but now it cant be more than an int? Not that I plan to use more than 2 Billion points anytime soon though 
Or make `Restart_Vars` above a `new unsigned long`

Reply:
These days I just use the largest one, to avoid those "narrow conversion" warnings and so on.

Reply:
Ok. For nPointFile you are narrowing it to the version before by giving it an int value ... but then again this is not an issue

Reply:
What?

Reply:
Anywho, testcase has been added, thanks for keeping me honest :blush:

Reply:
You replace `geometry->GetnPointDomain()` (which returns an unsigned long) with `nPointFile` which is an unsigned long as well. So far so good. But `const unsigned long nPointFile = Restart_Vars[2];` where `Restart_Vars` is an array of ints, i.e. `nPointFile` cannot be larger than int -> that is the 'narrowing [to the code] version of before' which I mean.

Reply:
Ah ok. I was thinking of this LGTM warning:
`
fixed alerts:
1 for Comparison of narrow type with wide type in loop condition
`
Comparison... I wrote conversion... doh! Same difference though, narrowing would be converting nPointFile to something shorter than int, nothing is lost by going up.